pkg/controller/common.go

```diff
@@ -305,6 +308,64 @@ func (c *SharedState) deleteOperationQueue(volumeName string) {
 	lockable.Unlock()
 }
 
+func (c *SharedState) createReplicaRequestsQueue() {
+	heap.Init(&c.priorityReplicaRequestsQueue.queue)
+	heap.Init(&c.failedReplicaRequestsQueue.queue)
+}
+
+func (c *SharedState) generateReplicaAttachmentRoutines(ctx context.Context, azr azReconciler, namespace string) error {
+	klog.V(5).Infof("Begin dequeue from replica attachment queue for nodes")
```

Comment:
I think we should leverage using semaphores to prevent multiple instances of `generateReplicaAttachmentRoutines` running concurrently.

Reply:
See https://pkg.go.dev/golang.org/x/sync/semaphore.